### ISSUE TITLE
Swap image title and alt text for Pinterest shares

### ISSUE THREAD

**Issue by nutsandbolts** [warfare-plugins/social-warfare#371]
Currently if there is no Pinterest image set or the core plugin is in use without Pro, the visitor will be given a choice of all the images on the page. The description (if there isn't a custom one set) for each image is the title, with the alt text as a fallback. We need to swap this so the alt text is used first, with the image title used only if there is no alt text.

You can see what I'm saying here ([link](http://www.carolinescooking.com/avocado-paletas-mexican-avocado-ice-pops/)) - this is a user who has the free plugin only. In the Pinterest share popup, all the images are showing the title (except the one she hid manually with a description set). I have no idea if this is intentional or working incorrectly but we should definitely fix it.

**Comment by Nicholas-Cardot** [comment]
I'll look into this, but I don't think there's anything that we can do about this. The link that we are using to activate the Pinterest share is as follows:

onClick="var e=document.createElement(\'script\');e.setAttribute(\'type\',\'text/javascript\');e.setAttribute(\'charset\',\'UTF-8\');e.setAttribute(\'src\',\'//assets.pinterest.com/js/pinmarklet.js?r=\'+Math.random()*99999999);document.body.appendChild(e);"

This code simply calls in and activated Pinterest native javascript function (directly from their website) without the ability to pass along to it any title, description, images or anything. Instead, their JS does 100% of the work and populates the share options from whatever it finds on the page. If it's using the image titles, then that is simply how their JS is designed to work.

**Comment by nutsandbolts** [comment]
Okay, good to know. We may need to look into what method some other social plugins are using, because I think using the alt text is pretty typical for most others.
